aptos-move/aptos-aggregator/src/aggregator_extension.rs

```diff
@@ -189,14 +246,15 @@ impl Aggregator {
                 if self.value <= value {
                     self.value = subtraction(value, self.value)?;
                     self.state = AggregatorState::PositiveDelta;
+                    if self.value > self.limit {
+                        self.record_overflow(Some(self.value));
+                    }
                 } else {
                     self.value = subtraction(self.value, value)?;
```

Comment:
If self.value is -10, value is 100, limit is 50, then subtraction(value, self.value) will succeed, but there is still an overflow. We were not catching that overflow earlier. So, I added this if statement to catch the overflow and record it.

Reply:
I think you are responding to the wrong comment?
in this if branch, value is lower than self value and we have ?; in the end, but if ? were to ever fire, we wouldn't record anything. But it will actually never fire, I think, because we are subtracting a smaller value from a larger value.

The other if branch, you correctly recorded the overflow, but you didn't actually disallow the overflow to happen.

Reply:
oh I saw your comment there, so maybe I misunderstand here. What I was proposing though was this:self.value = subtraction(self.value, value).expect("Subtraction of smaller value from larger value must succeed")

Reply:
Done.